### PR TITLE
Sensitive identity exception details should not leak to the end user

### PR DESCRIPTION
**Pull request by abaniuszewicz**
After this change any sensitive exception data should not leak to the end user (incl. error code that could provide some hint to the potential hacker). Identity exceptions, unless explicitly defined that they can leak, will result in generic `400 Bad Request` status code with `Invalid request.` message.

### CODE REVIEW

**Comment by abaniuszewicz** [comment]
Also, added one missing feature around devices - user should not be able to refresh tokens from different device than the device for which the tokens were issued. Just added integration test for that on 0affd4e.

File: backend/DevUp/tests/DevUp.Api.Tests.Integration/V1/Identity/IdentityControllerTests.cs [hunk 1]
```diff
@@ -61,28 +59,48 @@ public async Task Register_WhenGivenInvalidRequest_ReturnsBadRequest()
...
         {
             // try to login without registering
             var result = await _apiClient.PostAsJsonAsync(Route.Api.V1.Identity.Login, _faker.LoginUserRequest);
-            var response = await result.Content.ReadFromJsonAsync<ErrorResponse>();
-
             result.Should().HaveStatusCode(HttpStatusCode.BadRequest);
-            response!.Errors.Should().ContainSingle(e => e == LoginException.InvalidUsernameMessage);
+
+            var response = await result.Content.ReadFromJsonAsync<ErrorResponse>();
+            response!.Errors.Should().ContainSingle(e => e == "Invalid request.");
```
Comment (Goush) [inline backend/DevUp/tests/DevUp.Api.Tests.Integration/V1/Identity/IdentityControllerTests.cs:103]: I'm not sure if that's the proper way of doing this. As a user I'm trying to log in, and I'm given `Invalid request` response, suggesting the request wasn't understood or correctly processed by servuur. I'd suggest using 401 (Unauthorized) code, with message: "Invalid credentials" - it's generic, doesn't provide information whether login, password or both are invalid.

**Review by Goush — CHANGES_REQUESTED** [review]

File: backend/DevUp/tests/DevUp.Api.Tests.Integration/V1/Identity/IdentityControllerTests.cs (hunk 1 above)
Comment (abaniuszewicz) [reply backend/DevUp/tests/DevUp.Api.Tests.Integration/V1/Identity/IdentityControllerTests.cs:103]: I think that code 400 is correct one here (unless we want to go deeper and use 422). 401 doesn't make much sense in this case, in my understanding it should be used when we want to access something and we're not authorized _yet_. In this particular case, login endpoint is not something that _requires_ user to be authorized, it's an endpoint that _grants_ authorization, therefore I'd stick to 400 (or 422, but that'd be for another pull request as change scope would have to be much greater).

File: backend/DevUp/src/DevUp.Api/V1/Middlewares/DomainErrorHandler.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+                await WriteErrorResponse(context, statusCode, response);
+            }
+            catch (DomainException exception)
+            {
+                var code = GetErrorCode(exception);
+                var response = new ErrorResponse(code, exception.Errors);
+                var statusCode = exception switch
+                {
+                    DomainDataValidationException => StatusCodes.Status400BadRequest,
```
[the remark this answers is not in the document]
Comment (abaniuszewicz) [reply backend/DevUp/src/DevUp.Api/V1/Middlewares/DomainErrorHandler.cs:30]: Yeah, that was kinda intentional here. I like to be explicit about these kind of things. I was thinking what status codes should be returned for different kind of failures and for now decided to stick to 400 with different error codes (i.e. exception names), but to leave a hint behind on what _can_ happen, I decided to leave explicit selection behind.

On the other hand, if you think that just leaving default fallback with `_ => 400` would be more readable, I can totally understand that and can apply the fix if you wish.

File: backend/DevUp/src/DevUp.Api/V1/Middlewares/ApplicationErrorHandler.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            {
+                await next(context);
+            }
+            catch (ApplicationException exception)
+            {
+                var code = GetErrorCode(exception);
+                var response = new ErrorResponse(code, exception.Errors);
+                var statusCode = exception switch
+                {
+                    ApplicationValidationException => StatusCodes.Status400BadRequest,
```
[the remark this answers is not in the document]
Comment (abaniuszewicz) [reply backend/DevUp/src/DevUp.Api/V1/Middlewares/ApplicationErrorHandler.cs:23]: Same as [other comment](https://github.com/abaniuszewicz/dev-up/pull/49#discussion_r990026441

File: backend/DevUp/src/DevUp.Api/V1/Middlewares/DomainErrorHandler.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Goush) [reply backend/DevUp/src/DevUp.Api/V1/Middlewares/DomainErrorHandler.cs:30]: I'm fine with that.

File: backend/DevUp/tests/DevUp.Api.Tests.Integration/V1/Identity/IdentityControllerTests.cs (hunk 1 above)
Comment (Goush) [reply backend/DevUp/tests/DevUp.Api.Tests.Integration/V1/Identity/IdentityControllerTests.cs:103]: Checked some sources, and couldn't find one answer - hence it seems like it's not standardized. We can leave it as is.

**Pull request merged**